doc/source/troubleshooting.rst

```diff
@@ -2,41 +2,72 @@
+Installation issues
+-------------------
 
-.. code-block:: bash
+Check progress with diagnose
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you're having trouble with a specific package, Shoebot provides a 'diagnose'
+command to check if things are working: ``python3 setup.py diagnose``
 
-    ** (gedit:3830): WARNING **: Could not load Gedit repository: Typelib file for namespace 'GtkSource', version '3.0' not found
+It's usually easiest to start with Python3 and Pycairo, then move on to
+PyGobject, Pango and Gtk3.
 
-or:
 
-.. code-block:: bash
+Try PGI with CairoCFFI and GTK3 instead of PyGobject and Pycairo
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Shoebot can run under PGI and CairoCFFI, which may be easier to install
+than the recommended setup with Pygobject and Pycairo.
 
-    ImportError: cannot import name Gedit
+In this setup Shoebot can work with the GUI, but text output is not available.
 
-then try installing the `gir1.2-gtksource-3.0` package:
+
+Other problems
+--------------
+
+TypeError: Couldn't find foreign struct converter for 'cairo.Context'
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you see this error, it means you're missing the Python GObject interface for
+cairo. On Debian/Ubuntu, this should be fixed with:
 
 .. code-block:: bash
 
-    sudo apt-get install gir1.2-gtksource-3.0
+    sudo apt-get install python3-gi-cairo
+
+See the :doc:`installation page <install>` to know the relevant dependencies in
+other distros.
 
-`This StackOverflow answer <http://askubuntu.com/a/414592>`_ helped on finding this solution. However, if you see this:
+The Gedit plugin does not activate
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Try running Gedit from the command line so that you can see debug messages. If
+you see one of these warnings:
 
 .. code-block:: bash
 
-   (gedit:6027): libpeas-WARNING **: Could not find loader 'python3' for plugin 'shoebotit'
+    ** (gedit:3830): WARNING **: Could not load Gedit repository: Typelib file for namespace 'GtkSource', version '3.0' not found
+
+or:
 
-It might be because you're using an outdated version of Gedit. We've found this issue on Gedit 3.4.x, and it disappeared after updating to version 3.8.
+.. code-block:: bash
 
+    ImportError: cannot import name Gedit
 
-TypeError: Couldn't find foreign struct converter for 'cairo.Context'
----------------------------------------------------------------------
+then try installing the ``gir1.2-gtksource-3.0`` package.
 
-If you see this error, it means you're missing the Python GObject interface for cairo. On Debian/Ubuntu, this should be fixed with:
+`This StackOverflow answer <http://askubuntu.com/a/414592>`_ helped on finding
+this solution. However, if you see this:
 
 .. code-block:: bash
 
-    sudo apt-get install python3-gi-cairo
+   (gedit:6027): libpeas-WARNING **: Could not find loader 'python3' for plugin 'shoebotit'
```

Comment:
Is this still an issue, Gedit 3.8 was from around 2014 ?

Reply:
happy to take out any old cruft!